copylot/hardware/orca_camera/camera.py

```diff
@@ -77,3 +125,337 @@ def run(self, nb_frame: int = 100000):
             )
 
         Dcamapi.uninit()
+
+    def live_capturing_and_show(self):
+        """
+        this function uses the orca camera for live capturing function, and display it in a cv2 window.
+
+        I'm now implementing some methods in a way that doesn't change anything in __init__() and run() that was already
+        there when I started.
+        Will re-organize these after discussion with AhmetCan.
+        -- Xiyu, 2022-11-16
+
+        highly possible we will delete this method after the discussion.
+        :return:
+        """
+        if Dcamapi.init():
+            dcam = Dcam(self._camera_index)
+            if dcam.dev_open():
+                nb_buffer_frames = 3
+                if dcam.buf_alloc(nb_buffer_frames):
+                    # dcamtest_thread_live(dcam)
+                    if dcam.cap_start():
+                        timeout_milisec = 100
+                        iwindowstatus = 0
+                        while iwindowstatus >= 0:
+                            if dcam.wait_capevent_frameready(timeout_milisec) is not False:
+                                data = dcam.buf_getlastframedata()
+                                iwindowstatus = dcamtest_show_framedata(data, 'test', iwindowstatus)
+                            else:
+                                dcamerr = dcam.lasterr()
+                                if dcamerr.is_timeout():
+                                    print('===: timeout')
+                                else:
+                                    print('-NG: Dcam.wait_event() fails with error {}'.format(dcamerr))
+                                    break
+
+                            key = cv2.waitKey(1)
+                            if key == ord('q') or key == ord('Q'):  # if 'q' was pressed with the live window, close it
+                                break
+
+                        dcam.cap_stop()
+
+                    dcam.buf_release()  # release buffer
+                else:
+                    print('-NG: Dcam.buf_alloc(3) fails with error {}'.format(dcam.lasterr()))
+                dcam.dev_close()
+
+        Dcamapi.uninit()
+
+    def live_capturing_return_images_get_ready(self, nb_buffer_frames=3, timeout_milisec=100):
```

Comment:
functions like this one are quite risky. Dcamapi is initialized here and who knows where it is uninitialized, these situations named as side-effects. We don't want them. We want to minimize the side-effects. DCAM API has many concepts that need init/uninit, so I am planning to implement these in a different way. For now you can remove this function and create a issue for the need so we can address it together @xiyuyi .